src/Axiom/Extensionality/Propositional.agda

```diff
@@ -44,3 +52,16 @@ lower-extensionality a₂ b₂ ext f≡g = cong (λ h → Level.lower ∘ h ∘
                    (∀ x → B₁ x) ≡ (∀ x → B₂ x)
 ∀-extensionality ext B₁ B₂ B₁≡B₂ with ext B₁≡B₂
 ∀-extensionality ext B .B  B₁≡B₂ | refl = refl
+
+-- Extensionality for explicit function spaces implies extensionality
+-- for implicit function spaces.
+
+implicit-extensionality : ∀ {a b} →
+                          Extensionality a b →
+                          ExtensionalityImplicit a b
+implicit-extensionality {a} {b} ext f≡g = cong explicit-to-implicit
+                                               (ext (λ x → f≡g))
+  where
+  explicit-to-implicit : {A : Set a}{B : A → Set b} →
+                         ((x : A) → B x) → ({x : A} → B x)
+  explicit-to-implicit f {x} = f x
```

Comment:
This seems like a useful combinator to have in `Function`.

Reply:
Any suggestion for a name? This is part of showing that `((x : A) → B x)` and `({x : A} → B x)` are equivalent, of course.

Reply:
Something like this maybe?

```agda
_$∙ : ∀ {a b} {A : Set a} {B : A → Set b} →
      (∀ x → B x) → (∀ {x} → B x)
f $∙ = f _
```

Reply:
Or rather with a dash to be consistent with `-,_` defined in `Data.Product`.

Reply:
`$-` seems like a reasonable idea. I've had to use it, and the converse (implicit to explicit), in some codes.